### DISCUSSION TITLE
Dump, Die, Debug

### DISCUSSION THREAD

**Discussion by zoispag** [spatie/laravel-ignition#16] [Q&A] [answered]
Hey there! 👋🏼

Following the [blogpost](https://freek.dev/2168-ignition-the-most-beautiful-error-page-for-laravel-and-php-got-a-major-redesign), I replaced the old `facade/ignition` dependency with `spatie/laravel-ignition` and my first instinct to see the new UI in action, was to type `ddd('hey there');` in my `web.php`, only to encounter that the helper does not exist anymore! 😰

```
Call to undefined function ddd()
```

Any chance we are "getting" the helper "back"?

**Comment by freekmurze** [comment, marked as answer]
We've never use that helper ourselves, and we removed that helper to simplify the package.

Should more people ask for `ddd`, we'll consider reading it 👍

**Reply by dostrog** [reply]
o tempora o mores… 

It was so cool when you've had anounced it https://flareapp.io/blog/1-introducing-ddd-a-new-global-helper-for-laravel.

Please, give ddd() `second` chance

**Reply by bradmanning** [reply]
Chiming in to request adding back `ddd` - calling this was my first instinct as well after switching to `spatie/laravel-ignition`.

**Comment by freekmurze** [comment]
Seems like a lot of people like `ddd`. I'd accept a PR on `spatie/ignition` that brings it back. If none such a PR comes in, I'll add this myself soon.

**Reply by AlexVanderbist** [reply]
Here's a working `ddd` helper for spatie/laravel-ignition on Laravel 9:

```php
if (! function_exists('ddd')) {
    function ddd()
    {
        $args = func_get_args();

        if (count($args) === 0) {
            throw new Exception('You should pass at least 1 argument to `ddd`');
        }

        call_user_func_array('dump', $args);

        $renderer = app()->make(\Spatie\LaravelIgnition\Renderers\ErrorPageRenderer::class);

        $exception = new Exception('Dump, Die, Debug');

        $renderer->render($exception);

        die();
    }
}
```
If anyone makes the PR, I'll happily merge it :)

**Reply by SirMishaa** [reply]
Does it have been merged ? I cannot found ddd for a standalone project :(

**Reply by freekmurze** [reply]
Yes, you can see it being defined here: https://github.com/spatie/laravel-ignition/blob/main/src/helpers.php
